### ISSUE TITLE
Unable to reproduce the paper results

### ISSUE THREAD

**Issue by mrlihellohorld** [TencentARC/BrushEdit#15]
from diffusers import StableDiffusionBrushNetPipeline, BrushNetModel, UniPCMultistepScheduler
import torch
import cv2
import numpy as np
from PIL import Image

base_model_path = "/home/sobey/lzq/BrushEdit-main/model/BrushEdit/base_model/realisticVisionV60B1_v51VAE"

brushnet_path = "/home/sobey/lzq/BrushEdit-main/model/Brushnet/segmentation_mask_brushnet_ckpt"

blended = False

image_path="/home/sobey/lzq/BrushEdit-main/examples/brushnet/src/test_image.jpg"
mask_path="/home/sobey/lzq/BrushEdit-main/examples/brushnet/src/test_mask.jpg"
caption="A cake on the table."

brushnet_conditioning_scale=1.0

brushnet = BrushNetModel.from_pretrained(brushnet_path, torch_dtype=torch.float16)
pipe = StableDiffusionBrushNetPipeline.from_pretrained(
    base_model_path, brushnet=brushnet, torch_dtype=torch.float16, low_cpu_mem_usage=False
)

pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)

pipe.enable_model_cpu_offload()

init_image = cv2.imread(image_path)[:,:,::-1]
mask_image = 1.*(cv2.imread(mask_path).sum(-1)>255)[:,:,np.newaxis]
init_image = init_image * (1-mask_image)

init_image = Image.fromarray(init_image.astype(np.uint8)).convert("RGB")
mask_image = Image.fromarray(mask_image.astype(np.uint8).repeat(3,-1)*255).convert("RGB")

generator = torch.Generator("cuda").manual_seed(1234)

image = pipe(
    caption, 
    init_image, 
    mask_image, 
    num_inference_steps=50, 
    generator=generator,
    brushnet_conditioning_scale=brushnet_conditioning_scale
).images[0]

if blended:
    image_np=np.array(image)
    init_image_np=cv2.imread(image_path)[:,:,::-1]
    mask_np = 1.*(cv2.imread(mask_path).sum(-1)>255)[:,:,np.newaxis]

    mask_blurred = cv2.GaussianBlur(mask_np*255, (21, 21), 0)/255
    mask_blurred = mask_blurred[:,:,np.newaxis]
    mask_np = 1-(1-mask_np) * (1-mask_blurred)

    image_pasted=init_image_np * (1-mask_np) + image_np*mask_np
    image_pasted=image_pasted.astype(image_np.dtype)
    image=Image.fromarray(image_pasted)

image.save("output.png")
![image](https://github.com/user-attachments/assets/165a79cc-8e25-4bfd-b1fa-2939538bab57)

**Comment by liyaowei-stu** [comment]
It seems that there is something wrong with your code.

1. We speculate that the reason for the unexpected artifact may be due to the mask's value under half-precision.
2. Please ensure that mixed precision is automatically handled using with **torch.autocast("cuda")**.
3. Ensure that the input images and masks for BrushNetPipe are correct. For this example, to replace the background of the cake, the input image and mask for the pipe should be cake_fg (where only the cake has values) and mask_bg (where the background is 255).

Here is my improved code.

```
from diffusers import StableDiffusionBrushNetPipeline, BrushNetModel, UniPCMultistepScheduler
import torch
import cv2
import numpy as np
from PIL import Image


if __name__ == "__main__":
    base_model_path = "./models/base_model/realisticVisionV60B1_v51VAE"
    brushnet_path = "./models/brushnetX/"
    blended = True

    image_path="./examples/brushnet/src/test_image.jpg"
    mask_path="./examples/brushnet/src/test_mask.jpg"
    caption="A cake on the table."

    brushnet_conditioning_scale=1.0

    brushnet = BrushNetModel.from_pretrained(brushnet_path, torch_dtype=torch.float16)
    pipe = StableDiffusionBrushNetPipeline.from_pretrained(
    base_model_path, brushnet=brushnet, torch_dtype=torch.float16, low_cpu_mem_usage=False
    )

    pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
    pipe.enable_model_cpu_offload()

    init_image = cv2.imread(image_path)[:,:,::-1]
    mask_image_fg = 1.*(cv2.imread(mask_path).sum(-1)>255)[:,:,np.newaxis]
    mask_image_bg = 1-mask_image_fg
    init_image_fg = init_image * mask_image_fg

    init_image_fg = Image.fromarray(init_image_fg.astype(np.uint8)).convert("RGB")
    mask_image_bg = Image.fromarray(mask_image_bg.astype(np.uint8).repeat(3,-1)*255).convert("RGB")

    generator = torch.Generator("cuda").manual_seed(1234)


    with torch.autocast("cuda"):    
        image = pipe(
            caption,
            init_image_fg,
            mask_image_bg,
            num_inference_steps=50,
            generator=generator,
            brushnet_conditioning_scale=brushnet_conditioning_scale
            ).images[0]

    if blended:
        image_np_ori = np.array(init_image)
        image_np_preds = np.array(image)

        mask_np_fg = 1.*(np.array(mask_image_fg).sum(-1)>255)[:,:,np.newaxis]
    
        mask_blurred_fg = cv2.GaussianBlur(mask_np_fg*255, (21, 21), 0)/255
        mask_blurred_fg = mask_blurred_fg[:,:,np.newaxis]
        mask_blurred_bg = 1-mask_blurred_fg


        image_pasted = image_np_preds * mask_blurred_bg + image_np_ori*mask_blurred_fg
        image_pasted=image_pasted.astype(np.uint8)
        image=Image.fromarray(image_pasted)
        flie_name = "blending_"+"_".join(caption.split(" "))
        image.save(flie_name+".jpg")
    else:
        image.save("_".join(caption.split(" "))+".jpg")
```

And results are as follows:

<img width="654" alt="image" src="https://github.com/user-attachments/assets/7eff0cf6-48f8-4342-9185-f9ad10971e15" />


If you have any questions, please contact us promptly.

**Comment by liyaowei-stu** [comment]
@mrlihellohorld 
Hi👋, have you solved this problem?

**Comment by mrlihellohorld** [comment]
> @mrlihellohorld Hi👋, have you solved this problem?

I still have problems when I run your code directly. I suspect it is a problem with the model. I will download the model again and communicate with you when I have the experimental results. Thank you very much.

**Comment by liyaowei-stu** [comment]
> > @mrlihellohorld Hi👋, have you solved this problem?
> 
> I still have problems when I run your code directly. I suspect it is a problem with the model. I will download the model again and communicate with you when I have the experimental results. Thank you very much.

Do you use the code I provided? 
And you can check whether it is the latest commit. If not, please execute git pull.
If you have any questions, please contact me in time.

**Comment by mrlihellohorld** [comment]
Thank you very much for your help. I made a stupid mistake, which was downloading the wrong model. I downloaded the model again and ran it with your code and the results were consistent with yours.

**Comment by liyaowei-stu** [comment]
> Thank you very much for your help. I made a stupid mistake, which was downloading the wrong model. I downloaded the model again and ran it with your code and the results were consistent with yours.

That's good, I will close this issue. Please contact me promptly if there are any questions.
